featuretools/entityset/entityset.py

```diff
@@ -865,14 +879,20 @@ def add_last_time_indexes(self, updated_entities=None):
                         lti_df.set_index(entity.index, inplace=True)
                         lti_df = lti_df.reindex(entity.last_time_index.index)
                         lti_df['last_time_old'] = entity.last_time_index
-                    if not isinstance(lti_df, dd.DataFrame) and lti_df.empty:
+                    if not (lti_is_dask or lti_is_koalas) and lti_df.empty:
                         # Pandas errors out if it tries to do fillna and then max on an empty dataframe
                         lti_df = pd.Series()
                     else:
-                        lti_df['last_time'] = lti_df['last_time'].astype('datetime64[ns]')
-                        lti_df['last_time_old'] = lti_df['last_time_old'].astype('datetime64[ns]')
-                        lti_df = lti_df.fillna(pd.to_datetime('1800-01-01 00:00')).max(axis=1)
-                        lti_df = lti_df.replace(pd.to_datetime('1800-01-01 00:00'), pd.NaT)
+                        if lti_is_koalas:
+                            lti_df['last_time'] = ks.to_datetime(lti_df['last_time'])
+                            lti_df['last_time_old'] = ks.to_datetime(lti_df['last_time_old'])
+                            # TODO: Figure out a workaround for fillna and replace
+                            lti_df = lti_df.max(axis=1)
```

Comment:
In testing on the instacart dataset for the predict next purchase problem, this line appears to be causing a performance issue.